adapters/finch/src/main/scala/caliban/FinchAdapter.scala

```diff
@@ -78,22 +78,35 @@ object FinchAdapter extends Endpoint.Module[Task] {
     skipValidation: Boolean = false,
     enableIntrospection: Boolean = true
   )(implicit runtime: Runtime[R]): Endpoint[Task, Json :+: Json :+: CNil] =
-    post(stringBody :: header("content-type")) { (body: String, contentType: String) =>
-      if (contentType == "application/graphql") {
-        val request = GraphQLRequest(Some(body))
-        executeRequest(request, interpreter, skipValidation = skipValidation, enableIntrospection = enableIntrospection)
-      } else {
-        val result = for {
-          query <- ZIO.fromEither(parse(body).flatMap(_.as[GraphQLRequest]))
-          result <- createRequest(
-                     query,
-                     interpreter,
-                     skipValidation = skipValidation,
-                     enableIntrospection = enableIntrospection
-                   )
-        } yield result
-        runtime.unsafeRunToFuture(result).future
-      }
+    post(queryParams :: stringBodyOption :: header("content-type")) {
+      (queryRequest: GraphQLRequest, body: Option[String], contentType: String) =>
+        (queryRequest, body, contentType) match {
+          case (queryRequest, _, _) if queryRequest.query.isDefined =>
+            executeRequest(
+              queryRequest,
+              interpreter,
+              skipValidation = skipValidation,
+              enableIntrospection = enableIntrospection
+            )
+          case (_, Some(_), contentType) if contentType == "application/graphql" =>
+            executeRequest(
+              GraphQLRequest(body),
+              interpreter,
+              skipValidation = skipValidation,
+              enableIntrospection = enableIntrospection
+            )
+          case (_, Some(bodyValue), contentType) if contentType == "application/json" =>
```

Comment:
If body is `None` or content type is not "application/json" or "application/graphql", this `match` clause will throw an exception. Can you either catch that case, or just use `case _ => ` for this one?

Reply:
Thanks for review comment.
Fix it, I think to treat the request as same as none query in the case.
If there is a feeling of wrongness, please tell me.